h/streamer/filter.py

```diff
@@ -34,8 +32,20 @@ class FilterHandler:
     def __init__(self, filter_json):
         self.filter = filter_json
 
-    def evaluate_clause(self, clause, target):
-        field_value = resolve_pointer(target, clause["field"], None)
+    def evaluate_clause(self, clause, annotation):
+        field_path = clause["field"]
+
+        # Extract the annotation property that corresponds to the "field"
+        # path in the filter. This only supports a small set of paths
+        # which are actually used by the Hypothesis client.
+        field_value = None
+        if field_path == "/uri":
+            field_value = annotation.target_uri
+        elif field_path == "/references":
+            field_value = annotation.references
+        elif field_path == "/id":
+            field_value = annotation.id
```

Comment:
Are these the only possible values for field_path? That could be handy info. Should there be a trailing `else raise` to catch other things?

Reply:
A client could specify other values for `field`, but in that case the filter will simply never match. This was also the behaviour previously if an unsupported `field` value was set - I kept that the same in this PR to limit the scope.

Reply:
The other thing we may want to add soon is to allow the client to configure which group(s) it is interested in, rather than just which URLs.